test/mod2pi.jl

```diff
@@ -194,3 +194,25 @@ testModPi()
 @test mod2pi(355.0f0) ≈ 3.1416228f0
 @test mod2pi(Int64(2)^60) == mod2pi(2.0^60)
 @test_throws ArgumentError mod2pi(Int64(2)^60-1)
+
+@testset "rem_pio2_kernel" begin
```

Comment:
I don't there there are sufficiently many test cases here. Do you have all possible branches covered here with at least one test point? I think that would be very useful.